Update Examples and Make Minor Changes

CoWinkKeyDinkInc opened the pull request:
This pull request updates some outdated examples and makes small changes to the documentation so it is more clear.

 - Added new portal example using two referenced regions in a self contained tag
 - Move the `game` attribute to `<map>` as a `<game>` tag
 - Replace `<include>` reference in Blitz with `<respawn auto="true"/>`
 - Remove references to `<ghostsquadron/>` and explain how it is adapted from the Blitz gamemode.
 - Minor spelling changes and fixes for clarification.

I was also thinking of readapting the sample ghostsquadron XML to a working usable template that can be mentioned in the examples section.

mitchts commented on docs/modules/objectives/other.mdx:
```diff
@@ -5,18 +5,16 @@ title: Other Gamemodes
-`NOTE:` This module must be used in conjunction with another gamemode.
+The ghostsquadron gamemode can be quicky adapted in conjunction with the Blitz gamemode.
+The default ghostsquadron classes and setup can be found [here](https://raw.githubusercontent.com/OvercastNetwork/maps.oc.tc/master/Blitz/GS/gs.xml).
+Ghost squadron maps are set with a custom gamemode title using the `<game>` tag.
```
The code that made ghost squadron what it was doesn't exist anymore (e.g. Demo's landmines, Tracker's compass, and Spider's spider-sense). This is what `<ghostquadron/>` did in the XML -- all the special class abilities that were unique to GS. This should just be removed altogether, in my opinion.

CoWinkKeyDinkInc replied:
I do remember the spider sense thing you were talking about, though it looks that tracker and spider were probably additional classes added on OCN that weren't included in the original map. We could probably cut down the unworking classes and make a nice working example.

CoWinkKeyDinkInc commented:
I removed the Ghost Squadron section and rearranged the order of the other gamemodes, since I think TDM and FFA might be more relevant for newer people looking at the documentation. I also updated a region example explaining filter orders.
